stubs/iterable.stub

```diff
@@ -369,3 +369,43 @@ class InfiniteIterator extends IteratorIterator {
      */
     public function key() {}
 }
+
+/**
+ * @template TKey
+ * @template TValue
+ * @template TIterator as Iterator<TKey, TValue>
+ *
+ * @template-implements OuterIterator<TKey, TValue>
+ * @template-implements ArrayAccess<TKey, TValue>
+ *
+ * @template-extends IteratorIterator<TKey, TValue, TIterator>
+ */
+class CachingIterator extends IteratorIterator implements OuterIterator, ArrayAccess, Countable {
+    const CALL_TOSTRING = 1 ;
+    const CATCH_GET_CHILD = 16 ;
+    const TOSTRING_USE_KEY = 2 ;
+    const TOSTRING_USE_CURRENT = 4 ;
+    const TOSTRING_USE_INNER = 8 ;
+    const FULL_CACHE = 256 ;
+
+    /**
+     * @param TIterator $iterator
+     * @param int-mask-of<self::*> $flags
+     */
+    public function __construct(Iterator $iterator, int $flags = self::CALL_TOSTRING) {}
+
+    /**
```

Comment:
weird indent here

Reply:
And the method `hasNext` might be totally unnecessary to stub too?

Reply:
Fixed.

Reply:
Going to remove the `hasNext` stub now: https://phpstan.org/r/6c660d5e-5394-4c7d-8a37-341815223542